x-pack/plugin/otel-data/src/main/resources/component-templates/ecs-tsdb@mappings.yaml

```diff
@@ -0,0 +1,12 @@
+version: ${xpack.oteldata.template.version}
+_meta:
+  description: Default mappings for the OpenTelemetry tsdb index template installed by x-pack
```

Comment:
Indeed it's orthogonal to OTel and could be reused by other tsdb templates.

So far my strategy was to only move stuff to core when it's shared across multiple templates - just to be on the safe side.

I don't have a strong opinion - both would work. @felixbarny any opinion on this?

Reply:
Perhaps an even simpler option: remove the component template for now, and define it inline in the `metrics-otel@template` index template. Then if a need to reuse it comes up later, we could extract it into something in core.

Reply:
The issue with moving this to `metrics-otel@template` is that `metrics-otel@custom` is defined after it. Since the first dynamic template that matches has precedence, user-defined custom templates in `metrics-otel@custom` would be ignored.

Therefore, we need to have a separate component template for these built-in dynamic templates that  go after `metrics-otel@custom`. I don't have a strong opinion on whether `ecs-tsdb@mappings` should be extracted to core. I don't think this component template will get re-used very soon. That's because for integrations, dimensions are defined explicitly. For OTel, all attributes and resource attributes are dimensions and therefore subject to the limitations of dimension fields. This component template takes these limitations into consideration, which is why it's very basic. Due to https://github.com/elastic/elasticsearch/issues/110387, we can't even map `*.ip` fields. So there's not much left of ECS that this component template is mapping.

I'm wondering if we can move the dynamic template to the `metrics-otel@template` index template. We'll need to test if adding dynamic templates to `metrics-otel@custom` will still have precedence.

Reply:
> We'll need to test if adding dynamic templates to `metrics-otel@custom` will still have precedence.

I looked into this, but I fail to test this.

With the current status of this PR (so nothing moved yet) I added `metrics-otel@custom` with another dynamic template (just as a test to map strings to integers):
```
"dynamic_templates": [
    {
      "all_strings_to_int": {
        "mapping": {
          "type": "integer"
        },
        "match_mapping_type": "string"
      }
    }
  ]
```

So far from what I see this has no effect - I'd have expected things `"attributes": { "foo": "1",}` to be mapped as a `1` integer.

So I already fail to verify if the current status allows to add dynamic mappings that have precedence over `ecs-tsdb@mappings`.

But I'm a bit confused anyway... What's even the use-case here? In  `ecs-tsdb@mappings` we map strings to keyword, because that's waht TSDB supports (it does not support strings) - so I wonder what could be even overwritten for this in `...@custom`.

In any case - I'm happy to look into this further, but maybe it'd be worth discussing this as a follow up.

Reply:
> But I'm a bit confused anyway... What's even the use-case here? In  ecs-tsdb@mappings we map strings to keyword, because that's waht TSDB supports (it does not support strings) - so I wonder what could be even overwritten for this in ...@custom.

One example could be that users want to map `*.myipfield` to the `ip` field type. But if we're mapping all strings to keyword fields in a dynamic template that is defined before the @custom component template, users can't do that.

Reply:
I've added a test, moved the dynamic template to the index template itself, and deleted `ecs-tsdb@mappings`: 16234d7671d1e9785c8f15d177ed632b535a08b5